spring-cloud-gcp-pubsub-stream-binder/src/test/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubMessageChannelBinderTests.java

```diff
@@ -0,0 +1,80 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.stream.binder.pubsub;
+
+import org.junit.BeforeClass;
+import org.junit.Rule;
+
+import org.springframework.cloud.gcp.stream.binder.pubsub.properties.PubSubConsumerProperties;
+import org.springframework.cloud.gcp.stream.binder.pubsub.properties.PubSubProducerProperties;
+import org.springframework.cloud.stream.binder.AbstractBinderTests;
+import org.springframework.cloud.stream.binder.ExtendedConsumerProperties;
+import org.springframework.cloud.stream.binder.ExtendedProducerProperties;
+import org.springframework.cloud.stream.binder.Spy;
+
+import static org.assertj.core.api.Assumptions.assumeThat;
+
+/**
+ * Integration tests that require the Pub/Sub emulator to be installed and only run if the
+ * GCP_PUBSUB_EMULATOR envvar is present.
+ *
+ * @author João André Martins
+ */
+public class PubSubMessageChannelBinderTests extends AbstractBinderTests<PubSubTestBinder,
+		ExtendedConsumerProperties<PubSubConsumerProperties>,
+		ExtendedProducerProperties<PubSubProducerProperties>> {
+
+	@Rule
```

Comment:
Nope, because each test provisions consumers/subscribers and we want a fresh environment for each test.

Reply:
??? Are you going to do that in the real life as well?
Fresh environment for each  consumer/subscriber?

Then tests are wrong and definitely has to clean up resources after themselves.

I use `binding.unbind();` to do that though.

It really would be much better and more closed to the real world if we have an external resource once and reuse it from each test.

Otherwise something is bad by design...

Reply:
It seems like all the tests [already clean after themselves](https://github.com/spring-cloud/spring-cloud-stream/blob/master/spring-cloud-stream-binder-test/src/main/java/org/springframework/cloud/stream/binder/AbstractBinderTests.java#L208).

Indeed, in real world, these tests should also be isolated IMO.

Reply:
No, my point that in real world you connect many applications with their own interests to the same Pub/Sub service and don't worry about cross-talking. Otherwise something is wrong from the applications perspective and has to be fixed.